Trino raises "End of file reached before reading fully" when reading parquet in JuiceFS

<!--
Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!
-->

**What happened**:
Trino often raises exceptions as quoted in title when reading parquet files, and Pread EOF records appears in access log accordingly. However the parquet files obtained from mounted folder are complete.

**What you expected to happen**:
Read fully and no exception on complete files like this.

**How to reproduce it (as minimally and precisely as possible)**:
Just query, it seems be unable to find a way to reproduce precisely.

**Anything else we need to know?**
Size of parquet files is about 10MB to 110MB.
Cannot find any juicefs log in log of trino.
Added some access logs to pread function in pkg/fs/fs.go, find that offset(some number Trino wants but less than file size) > File.info.Size()(always 4096) in EOF log lines, however OK lines appear previously.

**Environment**:
- JuiceFS version (use `juicefs --version`) or Hadoop Java SDK version: 1.0.4
- Cloud provider or hardware configuration running JuiceFS: HuaweiCloud
- OS (e.g `cat /etc/os-release`): EulerOS 2.0 (SP2)
- Kernel (e.g. `uname -a`): Linux 3.10.0 x86_64
- Object storage (cloud provider and region, or self maintained): HuaweiCloud OBS
- Metadata engine info (version, cloud provider managed or self maintained): PostgreSQL, HuaweiCloud RDS
- Network connectivity (JuiceFS to metadata engine, JuiceFS to object storage):
- Others:
[eof-log.zip](https://github.com/juicedata/juicefs/files/11247956/eof-log.zip)

> > > > It would be great if you can provide a way to reproduce this issue.
> > > > My guess is the Attr is modified by other threads, can you print all the fields of Attr when it failed?
> > > 
> > > 
> > > I add File.info.attr struct log using File.fs.log() in pread function when offset larger than f.info.Size(), is this enough?
> > 
> > 
> > yes
> 
> pread (/a/b/c/d/e.db/f/month=202208/day=9/part-00001-65217629-b31f-4229-ad9b-db812c4c6e93.c000.snappy.parquet,10016459,4096): (0,EOF) offset(4096) >= f.info.Size()(4096) Attr: &{Flags:0 Typ:2 Mode:511 Uid:0 Gid:0 Rdev:0 Atime:1671415951 Mtime:1676016154 Ctime:1676016154 Atimensec:872490000 Mtimensec:603499000 Ctimensec:603499000 Nlink:8 Length:4096 Parent:1 Full:true KeepCache:true}
> 
> this is the attr when it failed, indicates the length is only 4096, but in directory mounted by juicefsctl, the length of this file is 116937516, the table stores archive, so no other apps modify it.

except for "End of file" exception, sometimes it raises "not found" exception
`
Caused by: java.io.FileNotFoundException: jfs://juicefs/a/b/c/d/e.db/f/month=202208/day=5/part-00000-65217629-b31f-4229-ad9b-db812c4c6e93.c000.snappy.parquet: not found
	at io.juicefs.JuiceFileSystemImpl.error(JuiceFileSystemImpl.java:196)
	at io.juicefs.JuiceFileSystemImpl.open(JuiceFileSystemImpl.java:974)
	at org.apache.hadoop.fs.FilterFileSystem.open(FilterFileSystem.java:164)
	at org.apache.hadoop.fs.PrestoFileSystemCache$FileSystemWrapper.open(PrestoFileSystemCache.java:325)
	at org.apache.hadoop.fs.FileSystem.open(FileSystem.java:899)
	at io.prestosql.plugin.hive.parquet.ParquetPageSourceFactory.createParquetPageSource(ParquetPageSourceFactory.java:160)
	... 21 more
`
I tried to set dir-entry-cache, entry-cache, attr-cache, open-cache all to 0 on over half nodes, "end of file" and "not found" exceptions appear on nodes from both halves.

--- a/pkg/meta/base.go
+++ b/pkg/meta/base.go
@@ -1242,11 +1242,15 @@ func (m *baseMeta) GetAttr(ctx Context, inode Ino, attr *Attr) syscall.Errno {
 	defer m.timeit("GetAttr", time.Now())
 	var err syscall.Errno
 	if inode == RootInode {
+		// doGetAttr could overwrite the `attr` after timeout
+		var a Attr
 		e := utils.WithTimeout(func() error {
-			err = m.en.doGetAttr(ctx, inode, attr)
+			err = m.en.doGetAttr(ctx, inode, &a)
 			return nil
 		}, time.Millisecond*300)
-		if e != nil || err != 0 {
+		if e == nil && err == 0 {
+			*attr = a
+		} else {
 			err = 0
 			attr.Typ = TypeDirectory
 			attr.Mode = 0777

